src/leader_scheduler.rs

```diff
@@ -225,9 +226,87 @@ impl LeaderScheduler {
             .collect()
     }
 
+    fn generate_leader_schedule(
+        &mut self,
+        tick_height: u64,
+        bank: &Bank,
+        default_schedule: Vec<Pubkey>,
+    ) -> LeaderSchedule {
+        assert!(default_schedule.is_empty());
+        self.seed = Self::calculate_seed(tick_height);
+        let active_set = self.get_active_set(tick_height, &bank);
+        let ranked_active_set = Self::rank_active_set(bank, active_set.iter());
+
+        let schedule = {
+            if ranked_active_set.is_empty() {
```

Comment:
Interesting special case. @aeyakovenko, this needs to go into the design document.  Instead of a default schedule, the node might want to fly solo in this case. It's actually a nice way to operate when we turn off voting too.